engines/basic_resources/app/collections/basic_resources/memberships_collection.rb

```diff
@@ -0,0 +1,30 @@
+module BasicResources
+  # Fetches the list of memberships that belong to the user, in the given group
+  # if specified. All the memberships that belong to the user are returned,
+  # otherwise.
+  class MembershipsCollection
+
+    # Constructor
+    #
+    # @param user [User]
+    # @param group [Group]
+    def initialize(user, group)
+      @user = user
+      @group = group
+    end
+
+    # Returns the memberships of the user either scoped by the given group or
+    # all of them
+    #
+    # @return [ActiveRecord::Relation<Membership>]
+    def build
+      relation = Membership.where(user_id: user.id)
+      relation = relation.where(basic_resource_group_id: group.id) if group
```

Comment:
It's not `plus`. Having a `group_id` filter doesn't mean that you'll show a subset nor a superset of the collection you would get without any filter. It's a completely different collection.

Does this help? https://github.com/coopdevs/katuma/pull/91#discussion_r82506801